src/edu/csus/ecs/pc2/clics/API202306/CLICSContestState.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 1989-2025 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.clics.API202306;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import edu.csus.ecs.pc2.core.Utilities;
+import edu.csus.ecs.pc2.core.model.ContestInformation;
+import edu.csus.ecs.pc2.core.model.ContestTime;
+import edu.csus.ecs.pc2.core.model.FinalizeData;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+import edu.csus.ecs.pc2.core.util.IJSONTool;
+import edu.csus.ecs.pc2.services.core.JSONUtilities;
+
+/**
+ * CLICS Contest State
+ *
+ * @author John Buck
+ *
+ */
+// 2023-06 Spec specifically says to return a time or null, so always include all properties
+// But it also says the fields are optional, so, I assume, if it's null, that means the CCS
+// supports that property, which PC2 does.  If it did not support the property, then it would
+// not be included in the JSON.  (At least that is my interpretation -- JohnB)
+@JsonInclude(JsonInclude.Include.ALWAYS)
+public class CLICSContestState {
+
+    @JsonProperty
+    private String started;
+
+    @JsonProperty
+    private String frozen;
+
+    @JsonProperty
+    private String ended;
+
+    @JsonProperty
+    private String thawed;
+
+    @JsonProperty
+    private String finalized;
+
+    @JsonProperty
+    private String end_of_updates;
+
+    /*
+     * These are non-standard PC2 specific properties to support pausing the contest
+     * 'paused' is the CONTEST elapsed time at which the contest was paused.  If set, it means the contest
+     * is currently paused. (HH:MM:SS.uuu)
+     *
+     * 'resumed' is the real (wall) time the contest was last resumed (or started, including initial contest start).
+     * You can tell if the contest had been paused at some point if resumed != started (ISO time)
+     */
+    @JsonProperty
+    private String paused;
+
+    @JsonProperty
+    private String resumed;
+
+    /*
+     * Non-standard property supplying a client with the current contest time (HH:MM:SS.uuu)
+     */
+    @JsonProperty
+    private String contest_time;
+
+    /**
+     * Fill in properties for contest state as per 2023-06 spec
```

Comment:
I'm not sure, but I think the above sentence will cause JavaDoc warnings -- because it doesn't end with a period.  (I believe the rule for JavaDoc is that the first sentence -- defined as everything up to and including the first period -- is what is used as the "method summary", while the rest of the JavaDoc header comments are included in the "method detail".

Reply:
Added periods at the end of sentences.